### ISSUE TITLE
TPMS TireMoni TM-240

### ISSUE THREAD

**Issue by 72ka** [merbanan/rtl_433#2425]
Hi, I have this TPMS TireMoni TM-240 system: [TM-240](https://shop.tiremoni.com/en/tyre-pressure-monitoring-system-tiremoni-tm-240/package-deals/safety-package/a-1946)

It has 4 sensors, up to 12bar pressure. Each sensors is labeled by number from 1 to 4. No ID outside even inside. Maybe there is helpfull link to FCCID [https://fccid.io/T8QTD-1000A](https://fccid.io/T8QTD-1000A) 

I spent couple of days (!) reading this awesome project wiki, and I stuck with it. My findings:

The sensors transmitts OOK signal, everything else is only my estimation... I think I can catch ID and Wheel position. You can look here for my current findings. There are the sensors 3 and 4 (S3 and S4), where I made some measurement under different pressure conditions (0,1,2,3 and 4 bars).

[BITBENCH HERE](https://triq.net/bitbench#c=%7B205%7D555555554e55566696969699aa5a9a96659aa696555a65966540%20%5BS3%200bar%5D&c=%7B203%7Daaaaaaaa9caaacd52d352d4cd4ad3534cb354d2caab4cb2d4a8&c=%7B205%7Daaaaaaaaa72aab332b4b4b4cd52ab54b32cd534b2aad32cb2aa0&c=&c=%7B210%7D2aaaaaaaaa72aab332b4ad4cad52d4acb32cd534b2aad32ccaca0%20%5BS3%201bar%5D&c=%7B208%7Daaaaaaaaa9caaacb4ad534cb554ccb554cb354d2caab4cb4d2b4&c=%7B208%7Daaaaaaaaa9caaacacad4b4cb2d4d4b54ccb354d2caab4cb4aab4&c=&c=%7B207%7D5555555553955596a5a6aaa6aa9695659966a9a5955699655690%20%5BS3%202bar%5D&c=%7B208%7Daaaaaaaaa9caaacad2d2d5552d4acab34cb354d2caab4cb2b348&c=%7B208%7Daaaaaaaaa9caaacad2d2d555354accb34cb354d2caab4cb35548&c=&c=%7B209%7D5555555554e55565a56a5a656aa665aaa659aa696555a65a695a0%20%5BS3%203bar%5D&c=%7B207%7D555555555395559595aa69995a9a96a99966a9a5955699695568&c=%7B207%7D555555555395559a95a56a6a6a9596a69966a9a5955699696590&c=&c=%7B208%7Daaaaaaaaa9caaacb4ad4caab4d4b4ad2cd2b54d2caab4cb32b48%20%20%5BS3%204bar%5D&c=%7B207%7D555555555395559595aa69995a9a96a99966a9a5955699695568&c=%7B206%7D555555555395559695a99556aa9699a59a56a9a5955699699a90&c=%7B208%7Daaaaaaaaa9caaacb4ad4caaab54b4cd2cd2b54d2caab4cb4ad48&c=&c=%7B212%7Deaaaaaaaaa9caaad54ccb4cacd53352d2d32b54d2ccaaab2d52c8%20%20%5BS4%200bar%5D&c=%7B203%7D4055555539555aa99969955aa66a565a656a9a599555656aa50&c=%7B209%7D5555555554e55569aa656655a69a6a9aa995aa696655559656540&c=&c=%7B208%7Daaaaaaaaa9caaad4d4cd2aaacd354d34d2b354d2ccaaab2cb4d4%20%5BS4%201bar%5D&c=%7B208%7Daaaaaaaaa9caaad554ccaaacd532cd3352b354d2ccaaab2d2cd4&c=%7B208%7Daaaaaaaaa9caaad4d4cd2aab2d354d34d2b354d2ccaaab2d54d4&c=&c=%7B208%7Daaaaaaaaa9caaad34ccccb4acd354b34d2b354d2ccaaab2ad534%20%5BS4%202bar%5D&c=%7B208%7D552aaaaaa9caaad2cccd4b4cb532cb3352b354d2ccaaab2aad34&c=%7B208%7Daaaaaaaaa9caaad34ccccb4b2d354b34d2b354d2ccaaab2ab534&f=8h%20WHEELPOS%3A%202d%204h%204h%2012h%204h%208h%208h%20ID%3A%2024h%208h%20%20&a=Preamble&m=553&d=MC&co=8&cl=15&cw=8)

Anyway, I'm totally lost... I cant see any relations in bits to the pressure.... No zereos when all sensors are in zero, etc... I am afraid is encrypted... 

I've tried to find anything in non-coded, manchester coded and differential manchester... all three ways ends the same, no relations to the pressure.

In the attachment you can see for example the recordings for Sensor 4 in one zip file. It consist of file mes.txt where is the console output with my commands and the records itself.

[S3-4bar.zip](https://github.com/merbanan/rtl_433/files/10980796/S3-4bar.zip)

Thank you for the help.

**Comment by zuckschwerdt** [comment]
Quick notes: I'd venture to say that "wheel position" is a fluke, usually there is a fixed ID which is learned and assigned by the head unit.
You can check if MC or DMC is used on a longer stream of slightly changing messages: measure for a day, either just outside to allow for small temperature and pressure changes or very slowly letting air out. You will only get a consistent ramp of values with the right coding. Also you will spot the field where the pressure is (and maybe temp) with a slow ramp.

The files 4,5,8,13 you posted are clipping badly (too loud) they could be FSK. The files 9,10,11 are incomplete.

Only file 12 is sensible -- still clipping but there are two complete packets. The signal is PPM 500/1000 -- this is not your TPMS. So we still need a good sample of the TPMS.

**Comment by 72ka** [comment]
> Only file 12 is sensible -- still clipping but there are two complete packets. The signal is PPM 500/1000 -- this is not your TPMS. So we still need a good sample of the TPMS.

Maybe a stupid question... how to check if the sample is not clipped? And not too loud? Is this sample good? Thank you :)


[OK_g017_433.92M_1000k.zip](https://github.com/merbanan/rtl_433/files/10981574/OK_g017_433.92M_1000k.zip)

**Comment by zuckschwerdt** [comment]
> how to check if the sample is not clipped?

I will add that to Pdv in the next weeks. For now try the preview https://triq.org/spectrogram-next/
It shows clipping with red (also the wave should look like a sine, no flat areas) in the bottom waveform view, zoom in with mouse scroll.

**Comment by 72ka** [comment]
OK, changed gain (rtl_433 -S all -s 2048k -g 12) and now I think this signals are correct. Now I think is clear, that the signal is FSK.
Unfortunatelly, the results are the same, only the PCM gives the consistent results in ID and also with wheel number...
Can you please confirm, that the signal is OK?

And... what do you think of the decoding? For me, everything looks like manchester, it works, but.. the results are not logical for pressure or temperature.

[S3-0bar.zip](https://github.com/merbanan/rtl_433/files/10983832/S3-0bar.zip)

Thank you

**Comment by zuckschwerdt** [comment]
Yes, that looks good. FSK PCM 50 MC is exactly what we expect from a TPMS.
I guess the sync is maybe `5539` as that aligns nicely to 10 full bytes.
Not sure if/where the encryption/obfuscation is as usually you'd want to hide the ID, which is still visible here?

**Comment by 72ka** [comment]
> 

Yes, the ID is visible that's doesnt make sense for encryption. But now I found very interresting thing, it looks like the packets are separated for some "types"... check [THIS BITBENCH](https://triq.net/bitbench#c=%7B208%7Daaaaaaaaa9caaacb52d2d2d34d4b4ab2ccb354d2caab4cb52ac8%20%5BS3%200bar%5D&c=%7B208%7Daaaaaaaaa9caaacb52d2d2d3554b4cb2ccb354d2caab4cb2ccc8%20%5BS3%200bar%5D&c=%7B209%7D5555555554e55565a96969695aa5a6596659aa696555a65956640%20%5BS3%200bar%5D&c=%7B208%7Daaaaaaaaa9caaacb52d2d2d52d4b4ab2ccb354d2caab4cb34ac8%20%5BS3%200bar%5D&c=%7B209%7D5555555554e55565a5696969aaa5a5596659aa696555a65965640%20%5BS3%200bar%5D&c=%7B209%7D5555555554e55565a56969695aa5a5596659aa696555a65955640%20%5BS3%200bar%5D&c=%7B208%7Daaaaaaaaa9caaacb4ad2d2d2d54b4ab2ccb354d2caab4cb34ac8%20%5BS3%200bar%5D&c=&c=%7B209%7D5555555554e555656969a96a96a56659a659aa696555a6596a640%20%5BS3%200bar%5D&c=%7B208%7Daaaaaaaaa9caaacad2d352d54d4accb34cb354d2caab4cb2b4c8%20%5BS3%200bar%5D&c=%7B209%7D5555555554e555656969a96aaaa56559a659aa696555a65959640%20%5BS3%200bar%5D&c=%7B208%7Daaaaaaaaa9caaacacad352d4cd4acab34cb354d2caab4cb532c8%20%5BS3%200bar%5D&c=%7B209%7D5555555554e555656569a96a96a56559a659aa696555a65969640%20%5BS3%200bar%5D&c=%7B208%7Daaaaaaaaa9caaacacad352d54d4acab34cb354d2caab4cb2b2c8%20%5BS3%200bar%5D&c=%7B209%7D5555555554e555655969a96a5aa56559a659aa696555a65a99640%20%5BS3%200bar%5D&c=&c=%7B207%7Dd55555555395559a95a9699a9a9aaa699966a9a5955699659aa8%20%5BS3%201bar%5D&c=%7B206%7D5555555553955596a5a6aaa5aa9695659966a9a5955699665690%20%5BS3%202bar%5D&c=%7B207%7D555555555395559695a969995a999aaa9966a9a59556996a6968%20%5BS3%203bar%5D&c=%7B208%7Daaaaaaaaa9caaacb4ad4caab4d4b4ad2cd2b54d2caab4cb32b48%20%5BS3%204bar%5D&f=6h%20WHEEL%3A%202d%20TYPE%3A%202h%202b%204b%20VALUE%3A%208d%204b%204h%2010b%206h%20ID%3A%2024h%208h%20%20&a=Preamble&m=5539&d=MC&co=8&cl=15&cw=8) If I order them by "type" look at the other values... I see there some consistency.
So maybe the sensors sends in each packet only for example pressure, then temperaure, then whatever (and also in different intervals... there is no separation to odd or even packets)... I think this is possible.

So, my next task is to collect more data.

**Comment by gdt** [comment]
@72ka What's the status of this issue and next steps?

**Comment by 72ka** [comment]
> @72ka What's the status of this issue and next steps?

Hi, I spent a lot of time on this device. It looks the device sends some like full packets with some unidentified informations periodiccaly and the pressure or temperature packets only when some event occurs (pressure leak etc...). I have no more time for it.

**Comment by merbanan** [comment]
No further communication and not enough data for further analysis.
